conflation/trace_fetching/mapillary.py

```diff
@@ -198,181 +206,332 @@ def make_trace_data_requests(
+                        IMAGES_URL.format(",".join(image_ids), access_token)
+                    )
+                    images = [
+                        {  # Convert to seconds because filtering / map matching assumes time in seconds
+                            "time": img_obj["captured_at"] / 1000,
+                            "lon": img_obj["geometry"]["coordinates"][0],
+                            "lat": img_obj["geometry"]["coordinates"][1],
+                        }
+                        for img_obj in images_resp.json()["data"]
+                    ]
 
-    print("Keys: {}".format(list(sequences_by_id.keys())))
+                    # Mapillary returns their trace data in random chronological order, so we need to sort the images
+                    images = sorted(images, key=lambda x: x["time"])
 
-    # We don't care about the sequence IDs anymore (just using it as a method to group trace data), so we just return
-    # values
-    sequences = list(sequences_by_id.values())
+                    # Only process sequences that originated from this bbox. This prevents us from processing sequences
+                    # twice
+                    origin_lon, origin_lat = (
+                        images[0]["lon"],
+                        images[0]["lat"],
+                    )
 
-    # Mapillary returns their trace data in reverse chronological order (latest image first), so we reverse that
-    # back to get the order the images were taken, which is what map matching needs
-    [s.reverse() for s in sequences]
+                    # We need to check if this specific sequence is covered by this zoom 14 tile. We do this by getting
+                    # the lon, lat coordinates of this tile and checking the first coordinate of the sequence. This way
+                    # we prevent the same sequence being processed by multiple different threads / zoom 14 tiles
+                    upper_left_corner = get_lon_lat_from_tile(
+                        BBOX_SECTION_ZOOM, tile[0], tile[1]
+                    )
+                    lower_right_corner = get_lon_lat_from_tile(
+                        BBOX_SECTION_ZOOM, tile[0] + 1, tile[1] + 1
+                    )
+                    cur_tile_as_lon_lat = [
+                        upper_left_corner[0],
+                        lower_right_corner[1],
+                        lower_right_corner[0],
+                        upper_left_corner[1],
+                    ]
+                    if not is_within_bbox(
+                        origin_lon,
+                        origin_lat,
+                        cur_tile_as_lon_lat,
+                    ):
+                        # print(
+                        #     "@@@ MAPILLARY: Skipping seq b/c origin ({}, {}) not in bbox_section {}".format(
+                        #         origin_lon, origin_lat, cur_tile_as_lon_lat
+                        #     )
+                        # )
+                        continue
```

Comment:
This log was getting annoying so I disabled it for now. Still a TODO to add a `--debug` logging option for this script.

Reply:
Added a counter for both duplicate-skips and filtered-skips.